apps/website/src/content/docs/themeprovider.mdx

```diff
@@ -0,0 +1,140 @@
+  themeOptions={{
+    applyBackground: false, // or true
+  }}
+/>
+```
+
+## Scoping and nesting
+
+By default, `ThemeProvider` will only apply its theme to its descendants, i.e. components outside of the provider's subtree will not be affected.
+
+This is useful if you want to have multiple themes in your application, or if you want to use iTwinUI for only a portion of your application.
+
+`ThemeProvider`s can also be nested if you want to use a different theme for a portion of your application.
+
+```jsx
+<ThemeProvider theme='light'>
+  <Text>Uses light theme!</Text>
+
+  <ThemeProvider theme='dark'>
+    <Text>Uses dark theme!</Text>
+  </ThemeProvider>
+</ThemeProvider>
+```
+
+### Inheritance
+
+When using `theme="inherit"` (which is also the default value), `ThemeProvider` will inherit the theme from its ancestors, e.g. from a `ThemeProvider` higher up in the tree.
+
+The inherited theme can also come from an older version of iTwinUI or from standalone usage of [variables](variables). If no parent theme is found, it will fall back to `"light"`.
+
+This behavior is useful for packages that want to use iTwinUI components but don't want to force a theme on their users. The application using the package can then wrap its entire tree in a `ThemeProvider` to apply a theme, which will be inherited by the package's components.
+
+```jsx
+const PackageComponent = () => {
+  return (
+    <ThemeProvider>
+      <Text>Uses theme from application!</Text>
+    </ThemeProvider>
+  );
+};
+
+const Application = () => {
+  return (
+    <ThemeProvider theme='dark'>
+      <PackageComponent />
+    </ThemeProvider>
+  );
+};
+```
+
+## Portals
+
+By default, `ThemeProvider` will be also be re-used as a [portal](https://react.dev/reference/react-dom/createPortal) target for floating elements (e.g. [`Tooltip`](tooltip), [`Toast`](toast), [`DropdownMenu`](dropdownmenu), [`Dialog`](dialog), etc).
+
+If you want to specify a different element as the portal target for all components within a tree, the `portalContainer` prop can be used.
+
+When passing an element to this prop, it is recommended to use state. For example:
+
+```jsx
+const [myPortal, setMyPortal] = React.useState(null);
+
+<div ref={setMyPortal} />
+
+<ThemeProvider portalContainer={myPortal}>
+  …
+</ThemeProvider>
+```
+
+## CSS
+
+Starting with v3, applications are advised to manually import `styles.css` in their entrypoint:
+
+```js
+import '@itwin/itwinui-css/css/styles.css';
+```
+
+This works well for applications that already use iTwinUI v3, but it can be tricky if the application is still using an older version of iTwinUI and wants to consume a package that uses v3.
+
+To make this easier, `ThemeProvider` can automatically import `styles.css` if it is not already imported. By default, this is enabled when using [`theme='inherit'`](#inheritance).
+
+If you want to override this behavior, you can use the `importCss` prop:
+
+```jsx
+<ThemeProvider importCss={false}>…</ThemeProvider>
+```
```

Comment:
Should there be a `PropsTable`?

Reply:
no need here, it's just a worse version (from readability perspective) of what is already explained on the page